src/main/java/com/google/sps/data/GetGovAdaptor.java

```diff
@@ -0,0 +1,46 @@
+package com.google.sps.data;
+
+import java.io.BufferedReader;
+import java.io.DataOutputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.ArrayList;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+
+/** File to run the GET request in java before putting it in the main adaptor function. 
+ */
+public class GetGovAdaptor {
+
+  public static ArrayList<NSWGovAQDataPoint> getSiteInfo() throws Exception {
+    //creating a get request
+    URL url = new URL("https://data.airquality.nsw.gov.au/api/Data/get_SiteDetails");
+    HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+    connection.setRequestMethod("GET");
+    
+    connection.setRequestProperty("Content-Type", "application/json");
+    
+    int responseCode = connection.getResponseCode();
+    
+    ArrayList<NSWGovAQDataPoint> convertedlist = null;
+    if (responseCode == HttpURLConnection.HTTP_OK) {
+      BufferedReader br = new BufferedReader(new InputStreamReader(connection.getInputStream()));
+      StringBuilder response = new StringBuilder();
+      String responseLine; 
+      while ((responseLine = br.readLine()) != null) {
+        response.append(responseLine.trim());
+      }
+ 
+      String responseString = response.toString();
+      Gson gson = new Gson();
+      convertedlist = gson.fromJson(responseString, new TypeToken<ArrayList<GovSiteDetails>>() {}.getType());
```

Comment:
The type that you are converting the json to here is different to the type of `convertedList`, I imagine this will cause an error. I think GovSiteDetails is the correct type though.

Reply:
To clarify, I think the return type of the function and convertedList should be GovSiteDetails